src/placement/pl_map_common.c

```diff
@@ -466,8 +467,9 @@ pl_map_extend(struct pl_obj_layout *layout, d_list_t *extended_list)
 		new_shards[grp_idx].po_fseq = f_shard->fs_fseq;
 		new_shards[grp_idx].po_shard = f_shard->fs_shard_idx;
 		new_shards[grp_idx].po_target = f_shard->fs_tgt_id;
-		if (org_shard->po_fseq > f_shard->fs_shard_idx &&
-				org_shard->po_target != -1)
+		if (org_shard->po_target != -1 &&
+		    (is_pool_adding ||
+		     org_shard->po_fseq > f_shard->fs_shard_idx))
```

Comment:
It is not your patch issue, but I am wondering, it is "org_shard->po_fseq > f_shard->fs_shard_idx" or "org_shard->po_fseq > f_shard->fs_fseq" ?

Reply:
I think should be org_shard->po_fseq > f_shard->fs_fseq, @wangdi1 ?
But in pool extending (before rebuilding), the po_fseq is same between them, so I added is_pool_adding check.